sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -1222,5 +1224,119 @@ public virtual async Task<Response<byte[]>> GetRandomBytesAsync(int count, Cance
                 throw;
             }
         }
+
+        /// <summary>
+        /// Releases a key.
+        /// </summary>
+        /// <param name="name">The name of the key to release.</param>
+        /// <param name="target">The attestation assertion for the target of the key release.</param>
+        /// <param name="options">Optional <see cref="ReleaseKeyOptions"/> containing additional options to release a key.</param>
```

Comment:
nit: Isn't "Optional" implied here?

Reply:
Not anymore, but the nuance may be lost on customers. I'm actually wondering from @KrzysztofCwalina and @tg-msft here if, because `target` is required but technically (over the wire) part of what `ReleaseKeyOptions` represents (in swagger, `KeyReleaseParameters`), if `options` should be required and `target` a required constructor parameter, like so:

```csharp
client.ReleaseKey("key-name", new ReleaseKeyOptions("JWS token value"));
```

Reply:
As written, though, this idiomatically matches what we're doing in other languages.

/cc @maorleger

Reply:
For JS I currently have the same API

```ts
releaseKey(name: string, target: string, options?: ReleaseKeyOptions): Promise<ReleaseKeyResult>;
```

My reasoning is that we can always do the mapping to the shape the service expects (target in parameters), but it is clear from the shape of this API what is _actually_ required. Required positional parameters + optional options bag. What do you think?

Reply:
I'm fine with the API as declared currently in both .NET and JS. We can ask specifically at the archboard review now that "options" can be required and have required parameters, though I prefer the current declaration.